apps/vaporgui/RenderEventRouterGUI.cpp

```diff
@@ -0,0 +1,85 @@
+#include "RenderEventRouterGUI.h"
+#include "UWidget.h"
+#include <QScrollArea>
+#include "VContainer.h"
+
+// clang-format off
+const std::string RenderEventRouterGUI::VariablesTabName  = "Variables";
+const std::string RenderEventRouterGUI::AppearanceTabName = "Appearance";
+const std::string RenderEventRouterGUI::GeometryTabName   = "Geometry";
+const std::string RenderEventRouterGUI::AnnotationTabName = "Annotation";
+// clang-format on
+
+RenderEventRouterGUI::RenderEventRouterGUI(VAPoR::ControlExec *ce, string paramsType) : RenderEventRouter(ce, paramsType)
+{
+    connect(this, &QTabWidget::currentChanged, this, &RenderEventRouterGUI::tabChanged);
+}
+
+
+QWidget *RenderEventRouterGUI::AddSubtab(string title, UWidget *subtab)
+{
+    VContainer *container = new VContainer(subtab);
+    container->AddBottomStretch();
+    container->SetPadding(0, 6, 0, 0);
+
+    QScrollArea *scrollArea = new QScrollArea;
+    scrollArea->setHorizontalScrollBarPolicy(Qt::ScrollBarAlwaysOff);
+    scrollArea->setWidget(container);
+    scrollArea->setWidgetResizable(true);
+
+    addTab(scrollArea, QString::fromStdString(title));
+    _subtabs.push_back(subtab);
+
+    return scrollArea;
+}
+
+void RenderEventRouterGUI::_updateTab()
+{
+    auto *params = GetActiveParams();
+    auto *paramsMgr = _controlExec->GetParamsMgr();
+    auto *dataMgr = GetActiveDataMgr();
+    if (!(params && paramsMgr && dataMgr)) return;
+
+    // clang-format off
+    auto gp = getGUIStateParams();
+    bool hasTab = false;
+    const auto activeTab = QString::fromStdString(gp->ActiveTab());
+    for (int i = 0; i <= count(); i++) {
+        if (activeTab == tabText(i)) {
+            hasTab = true;
+            setTab(i);
+            break;
+        }
+    }
+    if (!hasTab) {
+        setTab(0);
+    }
+    // clang-format on
+
+    for (Updateable *subtab : _subtabs) subtab->Update(params, paramsMgr, dataMgr);
+}
+
+void RenderEventRouterGUI::setTab(int i)
+{
+    blockSignals(true);
+    setCurrentIndex(i);
+    blockSignals(false);
+
+    auto pm = _controlExec->GetParamsMgr();
+    bool undoEnabled = pm->GetSaveStateUndoEnabled();
+    pm->SetSaveStateUndoEnabled(false);
+    getGUIStateParams()->SetActiveTab(tabText(i).toStdString());
+    pm->SetSaveStateUndoEnabled(undoEnabled);
+}
+
+// clang-format off
```

Comment:
I'm assuming that these comments are directives to the clang formatter. There seems to be several instances where your disabling the formatter for what looks like pretty basic code. Is there a reason for this?

Reply:
After running the formatter I found that my original formatting was clearer, however I can undo that if you'd like.

Reply:
It would seem that we're defeating the purpose of using a formatter if we're continuously disabling it :-)

Reply:
Ok done.